.github/workflows/ci-testing.yml

```diff
@@ -267,25 +278,42 @@ jobs:
           if "${{ matrix.cxx_compiler }}" == "cl" (
             call "C:\Program Files (x86)\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
           )
-          powershell $output = cmake --version; Write-Host ::warning::CMake: $output
-          powershell $output = ${{ matrix.cxx_compiler }} --version; Write-Host ::warning::Compiler: $output
 
           if "${{ matrix.backend }}" == "dpcpp" (
             set ninja_targets="build-onedpl-sycl_iterator-tests"
             set ctest_flags=-R sycl_iterator_.*\.pass
-            echo ::warning::dpcpp backend is set. Compile and run only sycl_iterator tests
           ) else (
             set ninja_targets=build-onedpl-tests
           )
           mkdir build && cd build
 
-          cmake -G "Ninja" -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DCMAKE_CXX_STANDARD=${{ matrix.std }} -DCMAKE_CXX_COMPILER=${{ matrix.cxx_compiler }} -DONEDPL_BACKEND=${{ matrix.backend }} -DONEDPL_DEVICE_TYPE=${{ matrix.device_type }} .. || goto :short_circuit_fail
-          ninja -j 2 -v %ninja_targets% || goto :short_circuit_fail
-          ctest --timeout %TEST_TIMEOUT% -C ${{ matrix.build_type }} --output-on-failure %ctest_flags% || goto :short_circuit_fail
-          exit /b 0
-          :: modify the default behaviour of shell:cmd, which exits with the status of a last command, in order not to unintentially miss an error
-          :short_circuit_fail
-          exit /b %errorlevel%
+          :: Preserve the code of an unsuccessful command if any.
+          :: By default, CMD shell only reports the error level of the final command.
+          set exit_code=0
+
+          cmake -G "Ninja" -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DCMAKE_CXX_STANDARD=${{ matrix.std }} -DCMAKE_CXX_COMPILER=${{ matrix.cxx_compiler }} -DONEDPL_BACKEND=${{ matrix.backend }} -DONEDPL_DEVICE_TYPE=${{ matrix.device_type }} -DCMAKE_CXX_FLAGS="/W4" .. || set exit_code=%errorlevel%
+
+          ninja -j 2 -v %ninja_targets% > build.log 2>&1 || set exit_code=%errorlevel%
+          :: Display the results after executing all tests because "tee" alternative is unavailable in CMD.
+          type build.log
+
+          ctest --timeout %TEST_TIMEOUT% -C ${{ matrix.build_type }} --output-on-failure %ctest_flags% > ctest.log 2>&1 || set exit_code=%errorlevel%
+          type ctest.log
+
+          :: Generate a summary
+          for /f "usebackq tokens=*" %%i in (`powershell -command "(Get-CimInstance -ClassName Win32_OperatingSystem).Caption"`) do set os_name=%%i
+          for /f "usebackq tokens=*" %%i in (`powershell -command "cmake --version | Select-Object -First 1"`) do set cmake_version=%%i
+          for /f "usebackq tokens=*" %%i in (`powershell -command "${{ matrix.cxx_compiler }} --version | Select-Object -First 1"`) do set compiler_version=%%i
```

Comment:
I think we need a special case for `cl` compiler here, because it does not like `--version`, it returns an error and we get a blank spot.
It seems the best way to get the version there is just `cl` on its own.

Reply:
Plain `cl` will also dump the version info the stderr. Let me try redirecting the output into a file and then reading it as a workaround...